src/index.js

```diff
@@ -1,38 +1,219 @@
+    const temp = [];
+    let desc = '';
+    desc = target.parentElement.firstElementChild.lastElementChild.textContent;
+    this.tasksList = ToDoList.getLocalList();
+    console.log(`temp${temp}`);
+
+    this.tasksList.forEach((item, index) => {
+      if ((((item.description).trim().toString()) === (desc.trim().toString()))) {
+        this.tasksList.splice(index, 1);
+        console.log(`after delete ${temp}`);
+        ToDoList.updateLocalList(this.tasksList);
+      }
+    });
+
+    // re-arrange index after delete
+    this.tasksList = ToDoList.getLocalList();
+    if (this.tasksList.length !== 0) {
+      for (let i = 0; i < this.tasksList.length; i += 1) { this.tasksList[i].index = i; }
+    }
+    // update the local storage
+    ToDoList.updateLocalList(this.tasksList);
+
+    // remove the item from the UI
+    target.parentElement.remove();
+  }
+
+  addToDo = (task) => {
+    const taskElement = document.createElement('div');
+    taskElement.className = 'tasks';
+    taskElement.innerHTML = `
+    <div class="task-wrapper">
+    <input type="checkbox" name="completed" class="status">
+    <span>${task} </span>
+    </div>
+    <i class="fas fa-ellipsis-v"></i>
+    <i class="fas fa-trash-alt"></i>
+    `;
+    tasksSection.appendChild(taskElement);
+
+    const checkBoxes = document.querySelectorAll('.status');
+    checkBoxes.forEach((cb) => {
+      cb.addEventListener('click', () => {
+        console.log(cb.nextElementSibling.textContent);
+
+        cb.parentElement.parentElement.classList.toggle('task-bg');
+        // cb.parentElement.lastElementChild.classList.toggle('task-completed');
+        cb.nextElementSibling.classList.toggle('task-completed');
+        cb.parentElement.parentElement.lastElementChild.classList.toggle('delete-task-icon');
+        cb.parentElement.parentElement.lastElementChild.previousElementSibling.classList.toggle('edit-task-icon');
+
+        // change status of the completed status
+        for (let i = 0; i < this.tasksList.length; i += 1) {
+          const cbTemp = cb.parentElement.lastElementChild.innerHTML;
+          if ((this.tasksList[i].description.trim().toString()) === ((cbTemp).trim().toString())) {
+            this.tasksList[i].completed = !(this.tasksList[i].completed);
+            ToDoList.updateLocalList(this.tasksList);
+          }
+        }
+        /*
+
+        (this.tasksList).forEach(task =>{
+          const cbTemp = cb.parentElement.lastElementChild.innerHTML;
+        if ((((task.description).trim().toString()) === ((cbTemp).trim().toString()))){
+          task.completed = !(task.completed);
+          this.updateLocalList(this.tasksList);
+        }
+      }); */
+      });
+    });
+
+    // find the delete icons here
+    const deleteIcons = document.querySelectorAll('.fa-trash-alt');
+    deleteIcons.forEach((di) => {
+      di.addEventListener('click', (e) => {
+        ToDoList.deleteTask(e.target);
+      });
+    });
+  }
+}
+
```

Comment:
- [ ] It would be better if we have a function for editing task description. I could not see it